packages/p2p-v2/src/components/Modals/ShareAdsModal/ShareAdsModal.tsx

```diff
@@ -0,0 +1,158 @@
+
+    const formattedMessage = customMessage.replace(/nn/g, '\n\n');
+
+    useEffect(() => {
+        Modal.setAppElement('#v2_modal_root');
+    }, []);
+
+    const onCopy = (event: { stopPropagation: () => void }) => {
+        copyToClipboard(advertUrl);
+        setIsCopied(true);
+        event.stopPropagation();
+    };
+
+    const handleGenerateImage = async () => {
+        if (divRef.current) {
+            const p2pLogo = divRef.current.querySelector('.p2p-v2-share-ads-card__qr-icon');
+            if (p2pLogo) {
+                const canvas = await html2canvas(divRef.current, { allowTaint: true, useCORS: true });
+                const screenshot = canvas.toDataURL('image/png', 1.0);
+                const fileName = `${type}_${id}.png`;
+                const link = document.createElement('a');
+                link.download = fileName;
+                link.href = screenshot;
+                link.click();
+            }
+        }
+    };
+
+    const handleShareLink = () => {
+        navigator.share({
+            text: formattedMessage,
+        });
+    };
+
+    useEffect(() => {
+        let timeoutClipboard: ReturnType<typeof setTimeout>;
+        if (isCopied) {
+            timeoutClipboard = setTimeout(() => {
+                setIsCopied(false);
+            }, 2000);
+        }
+        return () => {
+            clearTimeout(timeoutClipboard);
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [isCopied]);
+
+    return (
+        <>
+            {!isLoadingInfo && (
+                <Modal
+                    className={clsx('p2p-v2-modal-styles p2p-v2-share-ads-modal p-[2.4rem] flex flex-col')}
+                    isOpen={isModalOpen}
+                    onRequestClose={onRequestClose}
+                    style={customStyles}
+                    testId='dt_p2p_v2_share_ads_modal'
+                >
+                    <Text weight='bold'>Share this ad</Text>
+                    {!isMobile && <Text className='pt-[2.6rem]'>Promote your ad by sharing the QR code and link.</Text>}
+                    <div className='flex flex-row p2p-v2-share-ads-modal__container'>
+                        <div className='flex flex-col p2p-v2-share-ads-modal__container__card'>
+                            <ShareMyAdsCard advert={advertInfo} advertUrl={advertUrl} ref={divRef} />
+                            <Button
+                                isFullWidth={isMobile}
+                                onClick={handleGenerateImage}
+                                textSize={isMobile ? 'md' : 'sm'}
+                                variant='outlined'
+                            >
+                                Download this QR code
+                            </Button>
+                            {isMobile && (
+                                <div className='flex w-full gap-[1rem] justify-between mt-[1.5rem]'>
```

Comment:
for gap-[1rem] you can just use `gap-4`,  and mt-[1.5rem] you can use `mt-6` instead 👍 lets add custom sizes if its not already there in tailwind 

```suggestion
                                <div className='flex w-full gap-4 justify-between mt-6'>
```